### PR TITLE
Kills BOTH /poison paths by turning poisonous into an element. (+fantasty prefix, sanity on attackingtarget signal, and more)

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Poisonous is now a component 🎉 

Giant spiders are now a subtype of hostile, as god intended
Bees are now a subtype of hostile and are singular in their type name, as god intended
Snakes are now a subtype of retaliate, as god intended

As poisonous is now able to be a fantasy prefix for guns, items, and creatures, it is a new prefix that generates fantasy names for a few types of toxins that you can roll. neat! 🤓

vampirism no longer applies to objects with no force, i know rpgloot has 0 balance but this was really silly considering vampirism is a flat heal

The attackingtarget signal now sends the result of the attack, which means more things that fail their attacks no longer proc their effects (lifesteal healing you if your attack doesn't register)

## Why It's Good For The Game

Holy fuck those paths were BAD! Also code improvements. Gee, I guess i'll justify the kinda feature part of this which is the fantasy prefix: It fits in PERFECTLY with the lifesteal, knockback, and other fun, wild, and powerful rpgloot things you can roll

## Changelog
:cl:
add: Fantasy Prefix for poisoned weapons
fix: Many things that would trigger on attacking someone even if you didn't successfully land a hit now only trigger when you do
spellcheck: fixed a few typos
refactor: Poisonous is an element, no more terrible pathing /poison/ woop woop
/:cl:

### CODE REVIEW

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -549,7 +549,8 @@
 #define COMSIG_LIVING_SET_BODY_POSITION  "living_set_body_position"
 ///From post-can inject check of syringe after attack (mob/user)
 #define COMSIG_LIVING_TRY_SYRINGE "living_try_syringe"
-
+///living_defense.dm attack_animal proc, send signal is from animal though (aka defense proc caller) (mob/user)
+#define COMSIG_LIVING_ANIMAL_ATTACK "living_animal_attack"
```
Comment (tralezab) [inline code/__DEFINES/dcs/signals.dm:553]: this needs to be cut

File: code/datums/elements/poisonous.dm [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/**
+ * Poisonous component; which makes the attacks of the simplemob attached poison the enemy.
+ *
+ * Used for spiders and bees!
+ */
+/datum/component/poisonous
```
Comment (Mothblocks) [inline code/datums/elements/poisonous.dm:6]: In datums/elements? ✔
Is stateless? ✔
Actually a component? ✔

Yep, it's gaming time.

File: code/datums/elements/poisonous.dm [hunk 3]
```diff
@@ -0,0 +1,62 @@
+/**
+ * Poisonous component; which makes the attacks of the simplemob attached poison the enemy.
```
Comment (Qustinnus) [inline code/datums/elements/poisonous.dm:2]: image


image

Comment (MrMelbert) [reply code/datums/elements/poisonous.dm:2]: image

File: code/datums/elements/poisonous.dm (hunk 2 above)
Comment (tralezab) [reply code/datums/elements/poisonous.dm:6]: Gotcha bitch

**Comment by Qustinnus** [comment]
image
https://www.youtube.com/watch?v=S02BHmWPZNs

**Comment by tralezab** [comment]
map diff renders failing is unrelated to my changes

**Comment by 81Denton** [comment]
> map diff renders failing is unrelated to my changes

I reran it, looks good now

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 4]
```diff
@@ -351,9 +351,10 @@
 
 
 /mob/living/simple_animal/hostile/proc/AttackingTarget(atom/attacked_target)
-	SEND_SIGNAL(src, COMSIG_HOSTILE_ATTACKINGTARGET, target)
 	in_melee = TRUE
-	return target.attack_animal(src)
+	var/result = target.attack_animal(src)
+	SEND_SIGNAL(src, COMSIG_HOSTILE_ATTACKINGTARGET, target, result)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm:356]: It's better if this is two separate signals--one for trying the attack (what it was before), and one for doing the attack (here).

This removes a needless branch parameter, but more importantly lets us easily implement cancelling as well.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
One minor nitpick.

**Comment by Mothblocks** [comment]
I like "try" and "do" better, but eh.

**Pull request merged**